frontend/src/v5/store/tickets/card/ticketsCard.selectors.ts

```diff
@@ -97,23 +99,82 @@ export const selectSelectedTemplate = createSelector(
 	selectTemplateById,
 );
 
-export const selectTicketPins = createSelector(
+export const selectFilteringCompleted = createSelector(
+	selectTicketsCardDomain,
+	(ticketCardState) => ticketCardState.filters.complete,
+);
+
+export const selectFilteringTemplates = createSelector(
+	selectTicketsCardDomain,
+	(ticketCardState) => ticketCardState.filters.templates,
+);
+
+export const selectFilteringQueries = createSelector(
+	selectTicketsCardDomain,
+	(ticketCardState) => ticketCardState.filters.queries,
+);
+
+export const selectTicketsFilteredByQueriesAndCompleted = createSelector(
 	selectCurrentTickets,
+	selectFilteringCompleted,
+	selectFilteringQueries,
+	selectCurrentTemplates,
+	(tickets, isComplete, queries, templates) => {
+		const filteredByCompleted = tickets.filter((ticket) => getTicketIsCompleted(ticket) === isComplete);
+		if (!queries.length) return filteredByCompleted;
+		return filteredByCompleted.filter((ticket) => {
+			const templateCode = templates.find((template) => template._id === ticket.type).code;
+			const ticketCode = `${templateCode}:${ticket.number}`;
+			return queries.some((q) => [ticketCode, ticket.title].some((str) => str.toLowerCase().includes(q.toLowerCase())));
+		});
+	},
```

Comment:
To avoid filtering twice the tickets we can use just one loop:

```suggestion
	(tickets, isComplete, queries, templates) => tickets.filter((ticket) => {
		const templateCode = templates.find((template) => template._id === ticket.type).code;
		const ticketCode = `${templateCode}:${ticket.number}`;
		return getTicketIsCompleted(ticket) === isComplete && 
		(!queries.length || queries.some((q) => [ticketCode, ticket.title].some((str) => str.toLowerCase().includes(q.toLowerCase()))));
	}),
```

Reply:
@carmenfan As it is now the tickets that has no issuproperties or safetibaseproperties are filetered out when filtering by complete. Is ok this or those tickets should appear regardless the value pf **completed** filter?

Reply:
yes this is fine for now